src/net.cpp

```diff
@@ -1737,7 +1737,10 @@ void CConnman::AcceptConnection(const ListenSocket& hListenSocket) {
     const CService addr_bind{MaybeFlipIPv6toCJDNS(GetBindAddress(*sock))};
 
     NetPermissionFlags permission_flags = NetPermissionFlags::None;
-    hListenSocket.AddSocketPermissionFlags(permission_flags);
+    auto it{m_listen_permissions.find(addr_bind)};
+    if (it != m_listen_permissions.end()) {
```

Comment:
fd81820214e695ba228a954506397c3d781fe3fe: do want to add an `Assume` here, given that `Bind` always adds an entry?

Reply:
Right, currently `Bind()` always adds an entry, but I do not want to impose that because the code here does not require it. It is fine if there is nothing in `m_listen_permissions`. Then the permissions will remain at `NetPermissionFlags::None`. So it is kind of optional for `Bind()` to add an entry to `m_listen_permissions` and no reason to enforce it.